backend/src/plugins/authorization/authorization.plugin.ts

```diff
@@ -24,17 +25,22 @@ const authorization: FastifyPluginAsync<Options> = async (fastify, opts) => {
       return;
     }
 
-    const [, token] = request.headers?.authorization?.split(' ') ?? [];
-    const { userId } = tokenService.decode<TokenPayload>(token);
+    try {
+      const [, token] = request.headers?.authorization?.split(' ') ?? [];
+      const { userId } = tokenService.decode<TokenPayload>(token);
 
-    const authorizedUser = await master.getMasterById(userId);
-    if (!authorizedUser) {
+      const authorizedUser = await master.getMasterById(userId);
```

Comment:
It is not always master. It could also be a worker. So, you also need to check if worker with giver userId exist

Reply:
agreed

Reply:
now we have a role in the token payload and specific method in the `auth` service. you can use it